### PR TITLE
Fix "max number of clients reached" errors and perf improvements

### PR DESCRIPTION
**Pull request by timgl**
## Changes
- Add broker_pool_limit=0 (see [explanation]( https://stackoverflow.com/questions/47106592/redis-connections-not-being-released-after-celery-task-is-complete))
- Merge in minor improvements in number of queries
- Cache Team calls for api token

[This should solves Daz's issues](https://github.com/PostHog/posthog/issues/581#issuecomment-629356390)

Before, Celery would be spinning up too many connections to Redis if many events were inserted.
See load test:
image
Error message:
image

## Load testing
With 1 Standard-1x web dyno I was able to achieve on average 20/sec
image

With 2 Standard-2x web dynos I was able to achieve 44/sec
image

After caching Teams, no real difference though lower response times and memory load
image

Each time, Redis would use about 21 connections so the advice would be to get a Redis server with slightly more connections.

## Checklist
- [ ] All querysets/queries filter by Team (if applicable)
- [ ] Backend tests (if applicable)

### CODE REVIEW

File: posthog/api/capture.py [hunk 1]
```diff
@@ -106,10 +107,14 @@ def get_event(request):
...
-    try:
-        team = Team.objects.only('pk').get(api_token=token)
-    except Team.DoesNotExist:
-        return cors_response(request, JsonResponse({'code': 'validation', 'message': "API key is incorrect. You can find your API key in the /setup page in PostHog."}, status=400))
+    team_id = TEAM_ID_CACHE.get(token)
+    if not team_id:
+        try:
+            team_id = Team.objects.only('pk').get(api_token=token).pk
+            if team_id:
+                TEAM_ID_CACHE[token] = team_id
```
Comment (EDsCODE) [inline posthog/api/capture.py:115]: does this cache ever get flushed?

File: posthog/celery.py [hunk 2]
```diff
@@ -20,6 +20,10 @@
 # Load task modules from all registered Django app configs.
 app.autodiscover_tasks()
 
+# Make sure Redis doesn't add too many connections
+# https://stackoverflow.com/questions/47106592/redis-connections-not-being-released-after-celery-task-is-complete
+app.conf.broker_pool_limit = 0
```
Comment (EDsCODE) [inline posthog/celery.py:25]: makes sense. curious if needing to open and close connections will become a bottleneck anyway but the load testing seems to hold up

**Review by EDsCODE — APPROVED** [review]
code lgtm. can't/not sure how to mimic the conditions for client failure locally

File: posthog/api/capture.py (hunk 1 above)
Comment (timgl) [reply posthog/api/capture.py:115]: It does when the server gets restarted. However the token is fairly static, I don't think there'd be many situations you'd change it

File: posthog/celery.py (hunk 2 above)
Comment (timgl) [reply posthog/celery.py:25]: Hm yes I think it does slow it down a little bit but at least it works, especially with the cheaper Redis instances

**Pull request merged**